default/python/AI/colonization/rate_pilots.py

```diff
@@ -1,26 +1,120 @@
 import freeOrionAIInterface as fo
+from logging import debug
 
+import AIDependencies
 from AIDependencies import Tags
-from freeorion_tools import get_species_tag_grade
+from colonization.claimed_stars import has_claimed_star
+from colonization.colony_score import MINIMUM_COLONY_SCORE, debug_rating, use_new_rating
+from common.fo_typing import SpeciesName
+from empire.pilot_rating import best_pilot_rating, medium_pilot_rating
+from freeorion_tools import (
+    get_species_tag_grade,
+    get_species_tag_value,
+    tech_is_complete,
+    tech_soon_available,
+)
 from freeorion_tools.caching import cache_for_session
 
-GOOD_PILOT_RATING = 4.0
-GREAT_PILOT_RATING = 6.0
-ULT_PILOT_RATING = 12.0
+GOOD_PILOT_RATING_OLD = 4.0
+GREAT_PILOT_RATING_OLD = 6.0
+ULT_PILOT_RATING_OLD = 12.0
 
-_pilot_tags_rating = {
+_pilot_tags_rating_old = {
     "NO": 1e-8,
     "BAD": 0.75,
+    "GOOD": GOOD_PILOT_RATING_OLD,
+    "GREAT": GREAT_PILOT_RATING_OLD,
+    "ULTIMATE": ULT_PILOT_RATING_OLD,
+}
+
+BAD_PILOT_RATING = 0.4
+GOOD_PILOT_RATING = 2.0
+GREAT_PILOT_RATING = 3.0
+ULT_PILOT_RATING = 4.0
+GOOD_SHIELD_BONUS = 0.3
+DETECTION_FACTOR = 0.08  # detection values -1 to 3
+FUEL_FACTOR = 0.1  # fuel values: -0.5 to 1.5
+# troop values range from 0.5 to 2 (or 3, but there is currently no species with ult. troops)
+# TROOP_FACTOR is not used by rate_piloting, only in rate_colony_for_pilots
+TROOP_FACTOR = 0.15
+
+_pilot_tags_rating = {
+    "BAD": BAD_PILOT_RATING,
     "GOOD": GOOD_PILOT_RATING,
     "GREAT": GREAT_PILOT_RATING,
     "ULTIMATE": ULT_PILOT_RATING,
 }
 
+_detection_tags_rating = {
+    "BAD": -1,
+    "GOOD": 1,
+    "GREAT": 2,
+    "ULTIMATE": 3,
+}
+
 
-@cache_for_session
 def rate_piloting_tag(species_name: str) -> float:
+    """
+    Wrapper while we have both rating functions.
+    This ensures survey_universe uses the same rating function like colonization.
+    """
+    if use_new_rating:
+        return rate_piloting(species_name)
+    return rate_piloting_old(species_name)
```

Comment:
Although this approach is shorter, I prefer the full version with else.  It's much easy to track execution paths, because code and logic are aligned.   

```suggestion
    if use_new_rating:
        return rate_piloting(species_name)
    else:
        return rate_piloting_old(species_name)
```